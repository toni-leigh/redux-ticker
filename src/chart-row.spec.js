import React from 'react';
import renderer from 'react-test-renderer';

import ChartRow from './chart-row';

const runTest = bidDiff => {
  expect(renderer.create(
    <ChartRow 
      bid={ 5000 }
      bidDiff={ bidDiff }
      timestampe={ 1234 }
    />
  ).toJSON()).toMatchSnapshot();
}

test('<ChartRow /> with positive change below arrow threshold', () => runTest(0.2));
test('<ChartRow /> with positive change above arrow threshold', () => runTest(2.0));
test('<ChartRow /> with negative change below arrow threshold', () => runTest(-0.2));
test('<ChartRow /> with negative change above arrow threshold', () => runTest(-2.0));
test('<ChartRow /> with no change', () => runTest(0));