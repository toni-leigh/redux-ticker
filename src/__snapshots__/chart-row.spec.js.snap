// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartRow /> with negative change above arrow threshold 1`] = `
<div
  className="chart-row chart-row--down"
>
  <span
    className="chart-row__bid"
  >
    5000
  </span>
  <span
    className="chart-row__extra"
  >
    [
    -2.0
    , 
    -0.040%
    ]
  </span>
  <span
    className="chart-row__arrows"
  >
    ↓↓↓↓↓↓
  </span>
</div>
`;

exports[`<ChartRow /> with negative change below arrow threshold 1`] = `
<div
  className="chart-row chart-row--down"
>
  <span
    className="chart-row__bid"
  >
    5000
  </span>
  <span
    className="chart-row__extra"
  >
    [
    -0.20
    , 
    -0.0040%
    ]
  </span>
  <span
    className="chart-row__arrows"
  >
    
  </span>
</div>
`;

exports[`<ChartRow /> with no change 1`] = `
<div
  className="chart-row chart-row--down"
>
  <span
    className="chart-row__bid"
  >
    5000
  </span>
  <span
    className="chart-row__extra"
  >
    [
    0.0
    , 
    0.0%
    ]
  </span>
  <span
    className="chart-row__arrows"
  >
    
  </span>
</div>
`;

exports[`<ChartRow /> with positive change above arrow threshold 1`] = `
<div
  className="chart-row chart-row--up"
>
  <span
    className="chart-row__bid"
  >
    5000
  </span>
  <span
    className="chart-row__extra"
  >
    [
    2.0
    , 
    0.040%
    ]
  </span>
  <span
    className="chart-row__arrows"
  >
    ↑↑↑↑↑↑
  </span>
</div>
`;

exports[`<ChartRow /> with positive change below arrow threshold 1`] = `
<div
  className="chart-row chart-row--up"
>
  <span
    className="chart-row__bid"
  >
    5000
  </span>
  <span
    className="chart-row__extra"
  >
    [
    0.20
    , 
    0.0040%
    ]
  </span>
  <span
    className="chart-row__arrows"
  >
    
  </span>
</div>
`;
